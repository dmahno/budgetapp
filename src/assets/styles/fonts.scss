@font-face {
  font-family: 'Golos Text';
  src:
    url('@/assets/fonts/GolosText-Regular.woff2') format('woff2'),
    url('@/assets/fonts/GolosText-Regular.ttf') format('ttf');
  font-weight: 400;
  font-style: normal;
}

@font-face {
  font-family: 'Golos Text';
  src:
    url('@/assets/fonts/GolosText-Medium.woff2') format('woff2'),
    url('@/assets/fonts/GolosText-Medium.ttf') format('truetype');
  font-weight: 500;
  font-style: normal;
}

@font-face {
  font-family: 'Golos Text';
  src:
    url('@/assets/fonts/GolosText-SemiBold.woff2') format('woff2'),
    url('@/assets/fonts/GolosText-SemiBold.ttf') format('truetype');
  font-weight: 600;
  font-style: normal;
}

@font-face {
  font-family: 'Golos Text';
  src:
    url('@/assets/fonts/GolosText-Bold.woff2') format('woff2'),
    url('@/assets/fonts/GolosText-Bold.ttf') format('truetype');
  font-weight: 700;
  font-style: normal;
}

@font-face {
  font-family: 'Golos Text';
  src:
    url('@/assets/fonts/GolosTextText-Black.woff2') format('woff2'),
    url('@/assets/fonts/GolosTextText-Black.ttf') format('truetype');
  font-weight: 800;
  font-style: normal;
}
