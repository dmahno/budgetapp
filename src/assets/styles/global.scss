@use 'assets/styles/fonts';
@use 'assets/styles/mobile';
@use 'assets/variables/colors';

* {
  box-sizing: border-box;
}

html,
body {
  margin: 0;
  padding: 0;
  size: 16px;
  font-family: 'Golos Text', serif;
  background: var(--backgroundColor);
  color: var(--textColor);
}

body {
  min-width: 320px;
  padding: 0 $indent-s;
}

@include mobile {
  body {
    font-size: 14px;
  }
}

h1,
h2,
h3,
h4,
h5,
h6,
p,
input,
button,
textarea,
ul,
li {
  margin: 0;
  padding: 0;
}

li {
  list-style-type: none;
}

.link,
.active {
  transition:
    background-color 0.2s ease,
    color 0.2s ease;
}

svg {
  fill: var(--iconColor);
}

a {
  color: var(--linkColor);
}

input:-webkit-autofill,
input:-webkit-autofill:focus {
  transition:
    background-color 0s 600000s,
    color 0s 600000s !important;
}

button {
  background-color: var(--buttonBackgroundColor);
  color: var(--buttonTextColor);
  border: none;
  padding: 10px 20px;
  cursor: pointer;
}

html {
  scroll-behavior: smooth;
}
