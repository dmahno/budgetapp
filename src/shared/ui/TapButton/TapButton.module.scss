.button {
  padding: $indent-s;
  font-size: $text-size-m;
  border: none;
  display: flex;
  flex-direction: column;
  position: relative;
  border-radius: 22px;
  border-top-left-radius: 12px;
  cursor: pointer;
  width: 180px;
  height: 120px;
  transition: background-color 0.2s ease;
  justify-content: space-between;

  &.primary {
    background-color: var(--buttonPrimaryBackgroundColor);
    color: var(--buttonPrimaryColor);
  }

  &.secondary {
    background-color: var(--buttonSecondaryBackgroundColor);
    color: var(--buttonSecondaryColor);
  }

  .iconWrapper {
    position: absolute;
    top: $indent-xs;
    right: $indent-xs;
    display: flex;

    .iconContainer {
      color: inherit;
    }

    svg {
      fill: currentColor;
    }
  }

  .textWrapper {
    align-self: flex-start;
  }
}
