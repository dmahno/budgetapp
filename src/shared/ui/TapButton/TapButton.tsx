import React from 'react';

import AddIcon from 'shared/assets/icons/add.svg';
import SubtractIcon from 'shared/assets/icons/totalAmount.svg';

import styles from './TapButton.module.scss';

interface ITapButtonProps {
  icon?: React.ComponentType<React.SVGProps<SVGSVGElement>> | null;
  text: string;
  onClick?: () => void;
  disabled?: boolean;
  appearance?: 'primary' | 'secondary';
}

export const TapButton: React.FC<ITapButtonProps> = ({
  icon: Icon,
  text,
  onClick,
  disabled,
  appearance = 'primary',
  ...props
}) => {
  const DefaultIcon =
    Icon !== undefined
      ? Icon
      : appearance === 'primary'
        ? AddIcon
        : SubtractIcon;

  return (
    <button
      disabled={disabled}
      className={`${styles.button} ${styles[appearance]}`}
      onClick={onClick}
      {...props}
    >
      {DefaultIcon && (
        <div className={styles.iconWrapper}>
          <DefaultIcon data-testid="icon" />
        </div>
      )}
      <div className={styles.textWrapper}>{text}</div>
    </button>
  );
};
