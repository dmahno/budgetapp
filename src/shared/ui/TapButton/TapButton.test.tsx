import {render, screen, fireEvent} from '@testing-library/react';
import {vi} from 'vitest';

import {TapButton} from './TapButton';
import styles from './TapButton.module.scss';

describe('TapButton Component', () => {
  it('renders with primary appearance by default', () => {
    render(<TapButton text="Primary Button" />);
    const button = screen.getByRole('button', {name: /primary button/i});
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass(styles.button);
    expect(button).toHaveClass(styles.primary);

    const icon = screen.getByTestId('icon');
    expect(icon).toBeInTheDocument();
  });

  it('renders with secondary appearance when specified', () => {
    render(<TapButton text="Secondary Button" appearance="secondary" />);
    const button = screen.getByRole('button', {name: /secondary button/i});
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass(styles.button);
    expect(button).toHaveClass(styles.secondary);

    const icon = screen.getByTestId('icon');
    expect(icon).toBeInTheDocument();
  });

  it('displays the correct default icon based on appearance', () => {
    const {rerender} = render(<TapButton text="Primary Button" />);
    const addIcon = screen.getByTestId('icon');
    expect(addIcon).toBeInTheDocument();

    rerender(<TapButton text="Secondary Button" appearance="secondary" />);
    const subtractIcon = screen.getByTestId('icon');
    expect(subtractIcon).toBeInTheDocument();
  });

  it('renders a custom icon when provided', () => {
    const CustomIcon = () => <svg data-testid="custom-icon" />;
    render(<TapButton text="Custom Icon Button" icon={CustomIcon} />);
    const customIcon = screen.getByTestId('custom-icon');
    expect(customIcon).toBeInTheDocument();
  });

  it('displays the correct text', () => {
    const buttonText = 'Click Me';
    render(<TapButton text={buttonText} />);
    const button = screen.getByRole('button', {name: buttonText});
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent(buttonText);
  });

  it('handles onClick events', () => {
    const handleClick = vi.fn();
    render(<TapButton text="Clickable Button" onClick={handleClick} />);
    const button = screen.getByRole('button', {name: /clickable button/i});
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('applies additional props correctly', () => {
    render(
      <TapButton
        text="Extra Props Button"
        data-testid="extra-props-button"
        disabled
        aria-label="Extra Props"
      />,
    );
    const button = screen.getByTestId('extra-props-button');
    expect(button).toBeDisabled();
    expect(button).toHaveAttribute('aria-label', 'Extra Props');
  });

  it('renders without an icon when none is provided and no default is selected', () => {
    render(<TapButton text="No Icon Button" icon={null} />);
    const icon = screen.queryByTestId('icon');
    expect(icon).not.toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const {asFragment} = render(<TapButton text="Snapshot Button" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
