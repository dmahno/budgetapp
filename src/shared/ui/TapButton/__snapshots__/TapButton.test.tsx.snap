// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TapButton Component > matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="_button_242c51 _primary_242c51"
  >
    <div
      class="_iconWrapper_242c51"
    >
      <svg
        data-testid="icon"
        fill="none"
        height="36"
        viewBox="0 0 36 36"
        width="36"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M19.5 9C19.5 8.17157 18.8284 7.5 18 7.5C17.1716 7.5 16.5 8.17157 16.5 9V16.5H9C8.17157 16.5 7.5 17.1716 7.5 18C7.5 18.8284 8.17157 19.5 9 19.5H16.5V27C16.5 27.8284 17.1716 28.5 18 28.5C18.8284 28.5 19.5 27.8284 19.5 27V19.5H27C27.8284 19.5 28.5 18.8284 28.5 18C28.5 17.1716 27.8284 16.5 27 16.5H19.5V9Z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="_textWrapper_242c51"
    >
      Snapshot Button
    </div>
  </button>
</DocumentFragment>
`;
